dxil_converter.cpp

```diff
@@ -5760,6 +5760,27 @@ void Converter::Impl::emit_execution_modes_post_code_generation()
 		builder().addCapability(spv::CapabilityDenormPreserve);
 		builder().addExecutionMode(spirv_module.get_entry_function(), spv::ExecutionModeDenormPreserve, 64);
 	}
+
+	// Opt into quad derivatives and maximal reconvergence for fragment shaders using
+	// QuadAll/QuadAny intrinsics to get meaningful behaviour for quad-uniform control
+	// flow, other quad ops are ignored for now.
+	if (options.supports_quad_control && execution_model == spv::ExecutionModelFragment && (
+			execution_mode_meta.waveops_include_helper_lanes || execution_mode_meta.needs_quad_derivatives))
```

Comment:
WaveOpsIncludeHelperLanes does not turn helper lanes on if they did not originally exist. What it does mean is that if helper lanes exist (due to derivative/quad ops being used), they must participate. Helper lanes can also flicker out of life if they are no longer needed in the shader.

Reply:
> For shaders using the new WaveOpsIncludeHelperLanes option, the helper lanes will be guaranteed to exist until the last wave operation in the shader code, in addition to quad and derivate ops. *The attribute will never activate any helper lanes*.